README.md

```diff
@@ -260,6 +260,22 @@ You can send commands to a terminal without opening its window by using the `ope
 
 see `:h expand()` for more details
 
+### Smart-Toggling
+
+When `:ToggleTerm` is used without a prefixed count, it defaults to "smart-toggling".
+This feature is designed to use your current neovim context to determine which terminal
+to open, thus helping to minimize the number of times you have to explicitly provide a count.
+
+Currently, terminal contexts exist on a per-tab basis. Each tab is "pinned" to
+the first terminal that is opened in it. If you call `:ToggleTerm` from a new tab,
+it defaults to opening, (and thus pinning) the last-opened terminal
+(which can be done explicitly with `:ToggleTermSmartLast`).
+Alternatively, you can call `:ToggleTermSmartNew` within a new tab to associate
```

Comment:
I think rather than a bunch of new commands which tbh I'm having a little trouble following the workflow for these it would be better if a user set a strategy for tabs or something, and then it just behaved like you intend without needing these extra commands

Reply:
The workflow I was envisioning was: I'm in a tab, I have a terminal associated with it. Now, when I open a new tab, I have a choice: either associate it with the same terminal (`:ToggleTerm`, which opens the last-used terminal if one isn't already associated) or a brand new one (`:ToggleTermNew`). Now, if my work in this new tab diverges to the point that I want to associate a different terminal with it, I can do `:ToggleTermContextClear` followed by `:ToggleTermNew` (or I can open some other existing terminal).